src/api/wayfire/signal-definitions.hpp

```diff
@@ -595,6 +595,23 @@ struct view_geometry_changed_signal : public _view_signal
     wf::geometry_t old_geometry;
 };
 
+/**
+ * name: view-change-workspace
+ * on: output
+ * when: Whenever the view's workspace changes. (Every plugin changing the
+ *   view's workspace should emit this signal).
+ */
+struct view_change_workspace_signal : public _view_signal
+{
+    wf::point_t from, to;
+
+    /**
+     * Indicates whether the old workspace is known.
+     * If false, then the `from` field should be ignored.
+     */
+    bool old_workspace_invalid = true;
```

Comment:
I realized this field name doesn't seem to make much sense. If the old workspace is invalid then shouldn't that mean that it should be ignored? Isn't this backwards? i.e.: this field should be called `old_workspace_valid`

Reply:
Ah, yeah, makes sense, nice catch! It should be `_valid` indeed.

Reply:
Done.